pkg/cli/admin/rebootmachineconfigpool/reboot_machine_config_pool.go

```diff
@@ -0,0 +1,200 @@
+	DynamicClient  dynamic.Interface
+
+	dryRun bool
+
+	Printer printers.ResourcePrinter
+
+	genericclioptions.IOStreams
+}
+
+func (r *RebootMachineConfigPoolRuntime) Run(ctx context.Context) error {
+	visitor := r.ResourceFinder.Do()
+
+	// TODO need to wire context through the visitorFns
+	err := visitor.Visit(r.rebootFromResourceInfo)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *RebootMachineConfigPoolRuntime) rebootFromResourceInfo(info *resource.Info, err error) error {
+	if err != nil {
+		return err
+	}
+
+	if machineConfigPoolKind != info.Object.GetObjectKind().GroupVersionKind() {
+		return fmt.Errorf("command must only be pointed at machineconfigpools")
+	}
+
+	uncastObj, ok := info.Object.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("not unstructured: %w", err)
+	}
+	machineConfigPool := &MachineConfigPool{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uncastObj.Object, machineConfigPool); err != nil {
+		return fmt.Errorf("not a secret: %w", err)
+	}
+
+	return r.rebootMachineConfigPool(context.TODO(), machineConfigPool)
+}
+
+func (r *RebootMachineConfigPoolRuntime) createOptions() metav1.CreateOptions {
+	if r.dryRun {
+		return metav1.CreateOptions{
+			DryRun:       []string{metav1.DryRunAll},
+			FieldManager: RebootMachineConfigPoolFieldManager,
+		}
+	}
+	return metav1.CreateOptions{
+		FieldManager: RebootMachineConfigPoolFieldManager,
+	}
+}
+
+func (r *RebootMachineConfigPoolRuntime) updateOptions() metav1.UpdateOptions {
+	if r.dryRun {
+		return metav1.UpdateOptions{
+			DryRun:       []string{metav1.DryRunAll},
+			FieldManager: RebootMachineConfigPoolFieldManager,
+		}
+	}
+	return metav1.UpdateOptions{
+		FieldManager: RebootMachineConfigPoolFieldManager,
+	}
+}
+
+func (r *RebootMachineConfigPoolRuntime) rebootMachineConfigPool(ctx context.Context, machineConfigPool *MachineConfigPool) error {
+	// do something better.
+	if machineConfigPool.Spec.MachineConfigSelector == nil {
+		return fmt.Errorf("machineconfigpools/%v cannot be rebooted with this command due to missing .spec.machineConfigSelector", machineConfigPool.Name)
+	}
+	if len(machineConfigPool.Spec.MachineConfigSelector.MatchExpressions) > 0 {
+		// TODO probably add an option to specify labels somehow?  Maybe its not common in the field?
+		return fmt.Errorf("machineconfigpools/%v cannot be rebooted with this command due to usage of .spec.machineConfigSelector.matchExpressions", machineConfigPool.Name)
+	}
+	if len(machineConfigPool.Spec.MachineConfigSelector.MatchLabels) == 0 {
+		return fmt.Errorf("machineconfigpools/%v cannot be rebooted with this command due to missing .spec.machineConfigSelector.matchLabels", machineConfigPool.Name)
+	}
+
+	machineConfig := restartTemplate.DeepCopy()
+	if machineConfig.Labels == nil {
+		machineConfig.Labels = map[string]string{}
+	}
+	machineConfig.Name = fmt.Sprintf("95-certrotation-rebooter-%s", machineConfigPool.Name)
```

Comment:
This isn't strictly scoped to cert rotation, so we could probably s/certrotation/oc-initiated-reboot/ maybe?

Reply:
Updated